### ISSUE TITLE
illegal base32 data at input byte 25

### ISSUE THREAD

**Issue by madtechsupport** [pcarrier/gauth#6]
Hi,

Please see below, an encrypted 26 character code fails to decode correctly.

Regards,

Warren

```
➜  ~  cat .config/gauth.csv
Test: abcdefghijklmnopqrstuvwxyz
➜  ~  gauth
           prev   curr   next
Test       434331 315822 272588
[==========                   ]
➜  ~  cp .config/gauth.csv gauth.plain
➜  ~  openssl enc -aes-128-cbc -md sha256 -in gauth.plain -out ~/.config/gauth.csv
enter aes-128-cbc encryption password:
Verifying - enter aes-128-cbc encryption password:
➜  ~  gauth
Encryption password:
           prev   curr   next
2015/12/17 02:43:31 illegal base32 data at input byte 25
```

**Comment by madtechsupport** [comment]
Hi, yes, it was a while ago but I think I first encountered the error when I tried to encrypt an AWS secret key. The example I provided above still works, I just downloaded and tested with go 1.9.2 and the latest version of gauth.

Also, I've just now tried to repeat the test using your github page example text. This was the result:

`$ cat .config/gauth.csv 
AWS:   ABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ234567
Airbnb:abcd efgh ijkl mnop
Google:a2b3c4d5e6f7g8h9
Github:234567qrstuvwxyz
$ go/bin/gauth
           prev   curr   next
AWS        400639 624036 629171
Airbnb     736343 779575 332308
2018/01/13 10:03:09 illegal base32 data at input byte 13
$ cat -vte .config/gauth.csv 
AWS:   ABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ234567$
Airbnb:abcd efgh ijkl mnop$
Google:a2b3c4d5e6f7g8h9$
Github:234567qrstuvwxyz$
$ vi .config/gauth.csv 
$ cat .config/gauth.csv 
AWS:   ABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ234567
Airbnb:abcd efgh ijkl mnop
$ go/bin/gauth 
           prev   curr   next
AWS        624036 629171 574747
Airbnb     779575 332308 150787
[=========================    ]
$ cp .config/gauth.csv gauth.plain
$ openssl enc -aes-128-cbc -md sha256 -in gauth.plain -out ~/.config/gauth.csv
enter aes-128-cbc encryption password:
Verifying - enter aes-128-cbc encryption password:
$ go/bin/gauth 
Encryption password: 
           prev   curr   next
AWS        777403 055918 761431
Airbnb     927045 249056 039059
[==                           ]`

So the above example is pretty odd, now I'm getting the illegal base32 for a non-encrypted string, but encrypting the longer AWS example string works fine.

Regards,

Warren.

**Comment by fabaff** [comment]
I see the same issue with the example data.

```bash
2021/01/27 13:45:35 Generating codes: illegal base32 data at input byte 13
```

```bash
$ cat ~/.config/gauth.csv
AWS:   ABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ234567
Airbnb:abcd efgh ijkl mnop
Google:a2b3c4d5e6f7g8h9
Github:234567qrstuvwxyz
```

**Comment by creachadair** [comment]
@fabaff It sounds as if you might have an out-of-date version of `gauth` installed. Can you check when the version you're using was built? E.g.,

```sh
ls -l "$(which gauth)"
```

I am not able to reproduce this error with the current code in the repository.

**Comment by creachadair** [comment]
Aha, I see the problem. There are two issues: One, I was using the wrong test input; and two, there is an invalid example in the README file. The base32 alphabet doesn't allow 8 and 9. I will fix the broken example (#40).

**Comment by pcarrier** [comment]
Wow nice snipe @creachadair sorry about that extra confusion.
